package model.proto;

option java_outer_classname = "Message";

message Msg{
     optional int32 header = 1;
     optional int32 version = 2;
     optional MsgData msgdata=3;
}

message MsgData{
  repeated IntItem intItem = 1;
  repeated LongItem longItem = 2;
  repeated FloatItem floatItem = 3;
  repeated DoubleItem doubleItem = 4;
  repeated StringItem stringItem = 5;
  repeated MessageItem messageItem = 6;
}

message IntItem{
    optional string key = 1;
    optional int32 value = 2;
}

message LongItem{
    optional string key = 1;
    optional int64 value = 2;
}

message FloatItem{
    optional string key = 1;
    optional float value = 2;
}

message DoubleItem{
    optional string key = 1;
    optional double value = 2;
}

message StringItem{
    optional string key = 1;
    optional string value = 2;
}

message MessageItem{
    optional string key = 1;
    optional MsgData msgdata = 2;
}


